kube/src/client/config_ext.rs

```diff
@@ -0,0 +1,86 @@
+use std::convert::TryFrom;
+
+use tower::util::Either;
+
+#[cfg(any(feature = "native-tls", feature = "rustls-tls"))] use super::tls;
+use super::{
+    auth::{AddAuthorizationLayer, RefreshingTokenLayer},
+    Auth,
+};
+use crate::{Config, Result};
+
+/// Extensions to `Config` for `Client`.
+pub trait ConfigExt {
```

Comment:
This seems to be a variably-sized trait based on features. That looks difficult to implement outside of kube. Is the plan for this to become public at some point?

Reply:
No, I'll seal it and document it.